### PR TITLE
chore: RHIF-129 - Adapt to using useChrome

### PR DESCRIPTION
**Pull request by bastilian**
# Description

Associated Jira ticket: # (issue)

Please include a summary of the change, what this fixes/creates/improves.

# How to test the PR

Please include steps to test your PR.

# Before the change

# After the change

# Dependent work link

# Checklist:

- [ ] The commit message has the Jira ticket linked
- [ ] PR has a short description
- [ ] Screenshots before and after the change are added
- [ ] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

**Comment by jmbuscemi** [comment]
tagging @RedHatInsights/teamnado for their awareness and can help out with activation key functionality as needed.
our QE team can also run our IQE regression tests once this is in Stage

File: src/Components/Authentication/Authentication.js
Comment (gkarat) [inline src/Components/Authentication/Authentication.js:18]: nit-pick: this should add more bit of safety by preventing calling `undefined` as a func if `hideGlobalFilter` is not defined
```suggestion
    isSuccess && chrome?.hideGlobalFilter?.();
```

File: src/Components/Authentication/Authentication.js
Comment (gkarat) [inline src/Components/Authentication/Authentication.js]: Can't we return the Loading component instead? Is empty component a good solution for this case?

File: src/utils/platformServices.js
Comment (gkarat) [inline src/utils/platformServices.js:10]: the same suggestion here:
```suggestion
      return chrome?.auth?.getUser?.();
```

**Review by gkarat — COMMENTED** [review]
LGTM! Works well locally without any errors/warnings. I am not really acquainted to the app so the comments are really superficial. Probably a good idea to have another look from one of the active maintainers too

File: src/Components/Authentication/Authentication.js
Comment (bastilian) [reply src/Components/Authentication/Authentication.js]: It's now returning the `<Loading />` component.

**Review by lukegrn — CHANGES_REQUESTED** [review]
It looks like all of the mutation queries are failing (ex. delete, addadditionalrepos, create, etc) - I think something is wrong with how the token is being passed to those because the requests have `Bearer undefined` in them.

I also mentioned this in slack but will mention it here too - the hook tests shoud be updated to mock the `useUser` hook - instead of mocking `window.insights.chrome.user`. I imagine these still pass because that is what those hooks look at under the hood - but we should change it now in case the implementation details of chroming changes.

**Comment by bastilian** [comment]
@jlgre Thank you for taking a look! I went over it again and things should now be in order and working properly.

File: src/hooks/__tests__/useUser.test.js
Comment (lukegrn) [inline src/hooks/__tests__/useUser.test.js:34]: Does this mock need to stay if its done in `config/setupTests.js`?

File: src/hooks/useActivationKey.js
Comment (lukegrn) [inline src/hooks/useActivationKey.js:21]: Is there a reason these functions are curried instead of just being defined as `const getActivationKey = (token, keyName) => { ... }`?

I'm only going to make this comment in the one spot - but every hook looks like it is like this.

**Review by lukegrn — COMMENTED** [review]
A couple more comments. That mock looks good now, but the tests it affects (most things in the `hooks/__tests__/` dir should be able to have their `window.insights.chrome.auth` definitions removed.

File: src/hooks/__tests__/useUser.test.js
Comment (bastilian) [reply src/hooks/__tests__/useUser.test.js:34]: Yeah, in these tests there are expectations that the default mock in `setupTests.js` would not fulfil.

Comment (lukegrn) [reply src/hooks/__tests__/useUser.test.js:34]: Perfect - thanks!

File: src/hooks/useActivationKey.js
Comment (bastilian) [reply src/hooks/useActivationKey.js:21]: It is primarily to be consistent across `useQuery` and `useMutation` on how the token is passed. I tried a few things, but they would have been a bit more all over the place with `react-query` version 3. 

There would be a way with version 4, where the token could be passed along via the `meta` option, which would work for both by using some wrapper hook that injects chrome.

**Comment by bastilian** [comment]
@jlgre Oh, Yeah! I overlooked those `window.insights` definitions. I'll remove them!

File: src/hooks/useActivationKey.js
Comment (lukegrn) [reply src/hooks/useActivationKey.js:21]: Perfect - that makes sense. Thanks!

**Comment by bastilian** [comment]
@jlgre The PR to add the hook used here is now pending review. Once it is merged and a new package version has been pushed, i'll bump the package here and the test should pass and this PR should then be ready for a final review.

**Comment by lukegrn** [comment]
> @jlgre The PR to add the hook used here is now pending review. Once it is merged and a new package version has been pushed, i'll bump the package here and the test should pass and this PR should then be ready for a final review.

Ack - I'll look today. So the Travis check is expected to be failing right now?

**Review by lukegrn — APPROVED** [review]
This looks good to me providing that test failure is expected right now

**Comment by bastilian** [comment]
@jlgre This should now be ready to get merged. I did click through it after the utils package upgrade, but I'd feel more comfortable if someone more familiar would take a final quick look and merge it.

**Pull request merged**
